tests/features/DiscussionTest.php

```diff
@@ -0,0 +1,236 @@
+                    'discussable_type' => DiscussionTestModel::class,
+                ])->id,
+            'body' =>  $this->faker->text
+        ]);
+
+        $this->post(route('core.discussions.storeReply'), $reply->toArray())->assertStatus(201);
+    }
+
+    /** @test */
+    public function can_update_reply_to_discussion()
+    {
+        $reply = Reply::create([
+            'discussion_id' => factory(Discussion::class)->create([
+                    'discussable_id' => $this->testModel->id,
+                    'discussable_type' => DiscussionTestModel::class,
+                ])->id,
+            'body' =>  $this->faker->text
+        ]);
+
+        $reply->body = 'edited';
+
+        $this->patch(
+            route('core.discussions.updateReply', $reply->id, false),
+            $reply->toArray()
+        )->assertStatus(200);
+
+        $this->assertEquals($reply->fresh()->body, 'edited');
+    }
+
+    /** @test */
+    public function can_delete_reply_to_discussion()
+    {
+        $reply = Reply::create([
+            'discussion_id' => factory(Discussion::class)->create([
+                    'discussable_id' => $this->testModel->id,
+                    'discussable_type' => DiscussionTestModel::class,
+                ])->id,
+            'body' =>  $this->faker->text
+        ]);
+
+        $this->delete(route('core.discussions.destroyReply', $reply->id, false))
+            ->assertStatus(200);
+    }
+
+    /** @test */
+    public function can_only_act_on_own_reply()
+    {
+        $reply = Reply::create([
+            'discussion_id' => factory(Discussion::class)->create([
+                    'discussable_id' => $this->testModel->id,
+                    'discussable_type' => DiscussionTestModel::class,
+                ])->id,
+            'body' =>  $this->faker->text
+        ]);
+
+        $this->actingAs($this->createForeignUser());
+
+        $this->delete(route('core.discussions.destroyReply', $reply->id, false))
+            ->assertStatus(403);
+        
+        $reply->body = 'edited';
+
+        $this->patch(
+            route('core.discussions.updateReply', $reply->id, false),
+            $reply->toArray()
+        )->assertStatus(403);
+    }
+
+    /** @test */
+    public function can_toggle_reaction()
+    {
+        $reactionParams = collect([
+            'reactableId' => factory(Discussion::class)->create([
+                    'discussable_id' => $this->testModel->id,
+                    'discussable_type' => DiscussionTestModel::class,
+                ])->id,
+            'reactableType' => 'discussion',
+            'type' => Reactions::Clap,
+            'userId' => auth()->user()->id,
+        ]);
+
+        $this->post(route('core.discussions.react'), $reactionParams->toArray())->assertStatus(200);
+    }
+
+    private function model()
+    {
+        $this->createTestTable();
+
+        return DiscussionTestModel::create(['name' => 'discussable']);
```

Comment:
let's move this method in the setup method. You would not expect a `model()` method to create a model and something else